### DISCUSSION TITLE
How are Calls to OpenAI Made?

### DISCUSSION THREAD

**Discussion by sadovsd** [MaartenGr/BERTopic#1753] [Q&A]
Hello, I am new to software development so sorry if this is an obvious question. I was wondering how exactly calls are made to openai when the OpenAI representation model is defined. I know that for a prompt that includes alot of information like [KEYWORDS] [DOCUMENTS] can take a while to return the response. How does it return labels for 100+ categories in such a short time? I there concurrency happening anywhere?

**Comment by MaartenGr** [comment]
There is no concurrency happening since the calls to OpenAI are only made once for each topic. So even if you have millions of documents, if you only create 100 topics, only 100 calls will be made to OpenAI with aggregated statistics about the topics.
